presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/MetastoreUtil.java

```diff
@@ -437,7 +437,7 @@ else if (type instanceof TinyintType
         for (HiveColumnHandle partitionKey : partitionKeys) {
             String name = partitionKey.getName();
             Domain domain = effectivePredicate.getDomains().get().get(partitionKey);
-            if (domain != null && domain.isNullableSingleValue()) {
```

Comment:
would it be possible to add test for it to prevent future regressions? Take a look at `io.prestosql.plugin.hive.metastore.cache.TestCachingHiveMetastore`

Reply:
All tests in `io.prestosql.plugin.hive.metastore.cache.TestCachingHiveMetastore` call `getPartitionNamesByFilter` with   `partitionKeysFilter`=`TupleDomain.all()`
I didn't find a way to create an expected `TupleDomain` of a desired partition filter.

Reply:
How would a desired partition filter look like?

(would example here https://github.com/prestosql/presto/blob/8b7ce40f84fb0c38818709800fec537cb398ce52/presto-jmx/src/test/java/io/prestosql/plugin/jmx/TestJmxTableHandle.java#L41 be helpful?)

Reply:
To test `MetastoreUtil#computePartitionKeyFilter` I need to construct also `effectivePredicate`